opslevel/terraform_type_conversions.go

```diff
@@ -105,6 +106,56 @@ func MapValueToOpslevelJson(ctx context.Context, mapValue basetypes.MapValue) (o
 	return mapAsJson, diags
 }
 
+// Converts an opslevel.FilterPredicate to a basetypes.ObjectValue
+// - both case sensitive fields need to be set by calling function
+func OpslevelFilterPredicateToObjectValue(ctx context.Context, filterPredicate *opslevel.FilterPredicate) basetypes.ObjectValue {
+	if filterPredicate == nil {
+		return types.ObjectNull(filterPredicateType)
+	}
+	predicateAttrValues := map[string]attr.Value{
+		"case_insensitive": types.BoolNull(),
+		"case_sensitive":   types.BoolNull(),
+		"key":              RequiredStringValue(string(filterPredicate.Key)),
+		"key_data":         OptionalStringValue(filterPredicate.KeyData),
+		"type":             RequiredStringValue(string(filterPredicate.Type)),
+		"value":            OptionalStringValue(filterPredicate.Value),
+	}
+	return types.ObjectValueMust(filterPredicateType, predicateAttrValues)
+}
+
+func FilterPredicateObjectToModel(ctx context.Context, filterPredicateObj basetypes.ObjectValue) (filterPredicateModel, diag.Diagnostics) {
+	var predicateModel filterPredicateModel
+
+	objOptions := basetypes.ObjectAsOptions{UnhandledNullAsEmpty: true, UnhandledUnknownAsEmpty: true}
+	diags := filterPredicateObj.As(ctx, &predicateModel, objOptions)
+	return predicateModel, diags
+}
+
+func ExtractFilterPredicateModel(ctx context.Context, predicateWantedAttrs map[string]attr.Value, givenModels []filterPredicateModel) (filterPredicateModel, diag.Diagnostics) {
```

Comment:
This function would be better off returning a normal go `error` so that the diags can instead be handled by the function that calls it.